### ISSUE TITLE
VAAPI GPU acceleration broken in 73.0.3664.3-0ubuntu1~ppa1~18.10.1, 73.0.3664.3-0ubuntu1~ppa2~18.10.1 and 74.0.3710.0-0ubuntu1~ppa6~18.10.1 (with Nvidia GPU?)

### ISSUE THREAD

**Issue by omen23** [saiarcot895/chromium-ubuntu-build#42]
VAAPI acceleration with NVIDIA GPUs doesnt work anymore with build 73.0.3664.3

```
Log Messages
[7914:7914:0110/172241.365928:WARNING:gpu_video_decode_accelerator.cc(383)] : Created the VDA factory
[7914:7914:0110/172241.365963:WARNING:gpu_video_decode_accelerator_factory.cc(246)] : Initializing VAAPI VDA.
[7914:7914:0110/172241.395425:WARNING:gpu_video_decode_accelerator.cc(397)] : Created VDA
[7914:7914:0110/172241.400165:ERROR:vaapi_wrapper.cc(1470)] : vaCreateConfig failed VA error: the requested VAProfile is not supported
[7914:7914:0110/172241.400217:ERROR:vaapi_wrapper.cc(851)] : Failed to create VaapiWrapper for va_profile: -1
[7914:7914:0110/172251.028977:WARNING:gpu_video_decode_accelerator.cc(383)] : Created the VDA factory
[7914:7914:0110/172251.029004:WARNING:gpu_video_decode_accelerator_factory.cc(246)] : Initializing VAAPI VDA.
[7914:7914:0110/172251.068072:WARNING:gpu_video_decode_accelerator.cc(397)] : Created VDA
[7914:7914:0110/172251.071247:ERROR:vaapi_wrapper.cc(1470)] : vaCreateConfig failed VA error: the requested VAProfile is not supported
[7914:7914:0110/172251.071299:ERROR:vaapi_wrapper.cc(851)] : Failed to create VaapiWrapper for va_profile: -1
[7914:7914:0110/172330.713441:WARNING:x11_util.cc(1425)] : X error received: serial 486, error_code 3 (BadWindow), request_code 4, minor_code 0 (Unknown)
[7914:8106:0110/172330.776564:WARNING:x11_util.cc(1425)] : X error received: serial 3628, error_code 3 (BadWindow), request_code 4, minor_code 0 (Unknown)
```

**Comment by Queuecumber** [comment]
I think I'm also seeing this, youtube videos refuse to play since the last version unless I disable HW acceleartion

**Comment by omen23** [comment]
I only know `GpuVideoDecoder` was changed to `MojoVideoDecoder` on ChromeOS - so the code had to be [patched ](https://github.com/saiarcot895/chromium-ubuntu-build/blob/master/debian/patches/enable_vaapi_on_linux_2.diff)

I dont have any insight on the code atm.

Kind Regards

**Comment by omen23** [comment]
I only know the default video decoder was changed from `GpuVideoDecoder` to `MojoVideoDecoder` on ChromeOS - so the code had to be [patched ](https://github.com/saiarcot895/chromium-ubuntu-build/blob/master/debian/patches/enable_vaapi_on_linux_2.diff)

I dont have any insight on the code atm.

Kind Regards

**Comment by saiarcot895** [comment]
Currently, only `GpuVideoDecoder` is being used, not `MojoVideoDecoder`.

Some fixes have been made upstream to allow using the GPU video decoder with `MojoVideoDecoder` (this was the intention, I think, but the way it was coded didn't allow for it).

**Comment by saiarcot895** [comment]
I suspect that [this](https://chromium.googlesource.com/chromium/src.git/+/6f429309308523dc26b0ff49c1acb2977dd5727d) change broke decoding, but can't confirm it yet.

**Comment by biswasab** [comment]
@saiarcot895 Actually, according to this [commit](https://github.com/chromium/chromium/commit/31225b9c5f3f685d65f742dc129241c30c32157c) , enable_vaapi=true will mojo by default on Linux builds. This was made to reduce the CL even lower and now you can only use_vaapi=true flag and with --ignore-gpu-blacklist anyone can enable gpu decode acceleration without add any extra flags/patch.

**Comment by saiarcot895** [comment]
The commit I posted earlier did indeed cause the breakage. In particular, I've narrowed it down to the return value of [this](https://chromium.googlesource.com/chromium/src.git/+/master/media/gpu/vaapi/vaapi_video_decode_accelerator.cc#105) function call. If this function returns true, then a slightly more optimized codepath is taken for the hardware decoding (I think? I'm not entirely sure on this).

For now, I'm making a new build with this function always returning false for Linux. I might be able to make something a bit more optimized later.

**Comment by saiarcot895** [comment]
Mojo will still remain disabled by default; I'll enable it by default once I can verify that things work through Mojo (at least on my machine).

**Comment by biswasab** [comment]
@saiarcot895 I'm running a mojo build as my default installation. It works fine. Here is the patch I use:
```
From 471cc613de285452262f4705d19df401950cd723 Mon Sep 17 00:00:00 2001
From: Akarshan Biswas <akarshan.biswas@gmail.com>
Date: Fri, 28 Dec 2018 12:40:31 +0530
Subject: [PATCH] Enable mojo vaapi when use_vaapi is true on Linux

---
 chrome/browser/about_flags.cc                |  6 +++---
 chrome/browser/flag_descriptions.cc          |  9 +++++++--
 chrome/browser/flag_descriptions.h           | 10 ++++++++--
 gpu/config/software_rendering_list.json      |  4 +++-
 media/media_options.gni                      |  9 ++++++---
 media/mojo/services/gpu_mojo_media_client.cc |  6 ++++--
 6 files changed, 31 insertions(+), 13 deletions(-)

diff --git a/chrome/browser/about_flags.cc b/chrome/browser/about_flags.cc
index db54ccb61..248872183 100644
--- a/chrome/browser/about_flags.cc
+++ b/chrome/browser/about_flags.cc
@@ -1797,7 +1797,7 @@ const FeatureEntry kFeatureEntries[] = {
         "disable-accelerated-video-decode",
         flag_descriptions::kAcceleratedVideoDecodeName,
         flag_descriptions::kAcceleratedVideoDecodeDescription,
-        kOsMac | kOsWin | kOsCrOS | kOsAndroid,
+        kOsMac | kOsWin | kOsCrOS | kOsAndroid | kOsLinux,
         SINGLE_DISABLE_VALUE_TYPE(switches::kDisableAcceleratedVideoDecode),
     },
 #if defined(OS_WIN)
@@ -2456,12 +2456,12 @@ const FeatureEntry kFeatureEntries[] = {
      FEATURE_VALUE_TYPE(service_manager::features::kXRSandbox)},
 #endif  // ENABLE_ISOLATED_XR_SERVICE
 #endif  // ENABLE_VR
-#if defined(OS_CHROMEOS)
+#if defined(OS_CHROMEOS) || defined(OS_LINUX)
     {"disable-accelerated-mjpeg-decode",
      flag_descriptions::kAcceleratedMjpegDecodeName,
      flag_descriptions::kAcceleratedMjpegDecodeDescription, kOsCrOS,
      SINGLE_DISABLE_VALUE_TYPE(switches::kDisableAcceleratedMjpegDecode)},
-#endif  // OS_CHROMEOS
+#endif  // OS_CHROMEOS // OS_LINUX
     {"v8-cache-options", flag_descriptions::kV8CacheOptionsName,
      flag_descriptions::kV8CacheOptionsDescription, kOsAll,
      MULTI_VALUE_TYPE(kV8CacheOptionsChoices)},
diff --git a/chrome/browser/flag_descriptions.cc b/chrome/browser/flag_descriptions.cc
index 6cc51bbfb..943b28f66 100644
--- a/chrome/browser/flag_descriptions.cc
+++ b/chrome/browser/flag_descriptions.cc
@@ -2996,15 +2996,20 @@ const char kTextSuggestionsTouchBarDescription[] =
 
 #endif
 
-// Chrome OS -------------------------------------------------------------------
+// Chrome OS Linux-------------------------------------------------------------------
 
-#if defined(OS_CHROMEOS)
+#if defined(OS_CHROMEOS) || (defined(OS_LINUX) && !defined(OS_ANDROID))
 
 const char kAcceleratedMjpegDecodeName[] =
     "Hardware-accelerated mjpeg decode for captured frame";
 const char kAcceleratedMjpegDecodeDescription[] =
     "Enable hardware-accelerated mjpeg decode for captured frame where "
     "available.";
+#endif
+
+// Chrome OS --------------------------------------------------
+
+#if defined(OS_CHROMEOS)
 
 const char kAllowTouchpadThreeFingerClickName[] = "Touchpad three-finger-click";
 const char kAllowTouchpadThreeFingerClickDescription[] =
diff --git a/chrome/browser/flag_descriptions.h b/chrome/browser/flag_descriptions.h
index 944695782..16a3c985d 100644
--- a/chrome/browser/flag_descriptions.h
+++ b/chrome/browser/flag_descriptions.h
@@ -1824,13 +1824,19 @@ extern const char kPermissionPromptPersistenceToggleDescription[];
 
 #endif  // defined(OS_MACOSX)
 
-// Chrome OS ------------------------------------------------------------------
+// Chrome OS and Linux ------------------------------------------------------------------
 
-#if defined(OS_CHROMEOS)
+#if defined(OS_CHROMEOS) || (defined(OS_LINUX) && !defined(OS_ANDROID))
 
 extern const char kAcceleratedMjpegDecodeName[];
 extern const char kAcceleratedMjpegDecodeDescription[];
 
+#endif // defined(OS_CHROMEOS) || (defined(OS_LINUX) && !defined(OS_ANDROID))
+
+// Chrome OS ------------------------------------------------------------------------
+
+#if defined(OS_CHROMEOS)
+
 extern const char kAllowTouchpadThreeFingerClickName[];
 extern const char kAllowTouchpadThreeFingerClickDescription[];
 
diff --git a/gpu/config/software_rendering_list.json b/gpu/config/software_rendering_list.json
index cb6118b6a..a3ee63dba 100644
--- a/gpu/config/software_rendering_list.json
+++ b/gpu/config/software_rendering_list.json
@@ -371,11 +371,13 @@
     },
     {
       "id": 48,
-      "description": "Accelerated video decode is unavailable on Linux",
+      "description": "VA-API is not supported by NVIDIA",
       "cr_bugs": [137247],
       "os": {
         "type": "linux"
       },
+      "vendor_id": "0x10de",
+      "driver_vendor": "NVIDIA",
       "features": [
         "accelerated_video_decode"
       ]
diff --git a/media/media_options.gni b/media/media_options.gni
index 46eaa5818..6e338f651 100644
--- a/media/media_options.gni
+++ b/media/media_options.gni
@@ -5,6 +5,7 @@
 import("//build/config/chrome_build.gni")
 import("//build/config/chromecast_build.gni")
 import("//build/config/features.gni")
+import("//media/gpu/args.gni")
 import("//testing/libfuzzer/fuzzer_test.gni")
 
 # Do not expand this list without double-checking with OWNERS, this is a list of
@@ -129,8 +130,9 @@ declare_args() {
   # |mojo_media_services|). When enabled, selected mojo paths will be enabled in
   # the media pipeline and corresponding services will hosted in the selected
   # remote process (e.g. "utility" process, see |mojo_media_host|).
-  enable_mojo_media = is_android || is_chromecast || is_chromeos || is_mac ||
-                      is_win || enable_library_cdms
+  enable_mojo_media =
+      is_android || is_chromecast || is_chromeos || is_mac || is_win ||
+      enable_library_cdms || (is_desktop_linux && use_vaapi)
 
   # Enable the TestMojoMediaClient to be used in mojo MediaService. This is for
   # testing only and will override the default platform MojoMediaClient, if any.
@@ -200,7 +202,8 @@ if (enable_mojo_media) {
       ]
       _default_mojo_media_host = "gpu"
     }
-  } else if (is_chromeos || is_mac || is_win) {
+  } else if (is_chromeos || is_mac || is_win ||
+             (is_desktop_linux && use_vaapi)) {
     _default_mojo_media_services = [ "video_decoder" ]
     _default_mojo_media_host = "gpu"
   }
diff --git a/media/mojo/services/gpu_mojo_media_client.cc b/media/mojo/services/gpu_mojo_media_client.cc
index 43c387339..d35dd31b6 100644
--- a/media/mojo/services/gpu_mojo_media_client.cc
+++ b/media/mojo/services/gpu_mojo_media_client.cc
@@ -52,7 +52,7 @@ namespace media {
 namespace {
 
 #if defined(OS_ANDROID) || defined(OS_CHROMEOS) || defined(OS_MACOSX) || \
-    defined(OS_WIN)
+    defined(OS_WIN) || defined(OS_LINUX)
 gpu::CommandBufferStub* GetCommandBufferStub(
     base::WeakPtr<MediaGpuChannelManager> media_gpu_channel_manager,
     base::UnguessableToken channel_token,
@@ -146,7 +146,9 @@ std::unique_ptr<VideoDecoder> GpuMojoMediaClient::CreateVideoDecoder(
       android_overlay_factory_cb_, std::move(request_overlay_info_cb),
       std::make_unique<VideoFrameFactoryImpl>(gpu_task_runner_,
                                               std::move(get_stub_cb)));
-#elif defined(OS_CHROMEOS) || defined(OS_MACOSX) || defined(OS_WIN)
+#elif defined(OS_CHROMEOS) || defined(OS_MACOSX) || defined(OS_WIN) || \
+   defined(OS_LINUX)
+
 #if defined(OS_WIN)
   if (base::FeatureList::IsEnabled(kD3D11VideoDecoder)) {
     return D3D11VideoDecoder::Create(
-- 
2.20.1
```

**Comment by omen23** [comment]
rolling back to chromium-browser_73.0.3642.0-0ubuntu1_ppa4_18.10.1_amd64.deb
 from 73.0.3664.3-0ubuntu1~ppa2~18.10.1

about:gpu still errors out

Log Messages

   -
   [24195:24195:0115/173319.749984:WARNING:gpu_video_decode_accelerator.cc(383)]
   : Created the VDA factory
   -
   [24195:24195:0115/173319.750024:WARNING:gpu_video_decode_accelerator_factory.cc(246)]
   : Initializing VAAPI VDA.
   -
   [24195:24195:0115/173319.808267:WARNING:gpu_video_decode_accelerator.cc(397)]
   : Created VDA
   - [24195:24195:0115/173319.812036:ERROR:vaapi_wrapper.cc(1467)] :
vaCreateConfig
   failed VA error: the requested VAProfile is not supported
   - [24195:24195:0115/173319.812112:ERROR:vaapi_wrapper.cc(848)] : Failed
   to create VaapiWrapper for va_profile: -1
   -
   [24195:24195:0115/173332.006357:WARNING:gpu_video_decode_accelerator.cc(383)]
   : Created the VDA factory
   -
   [24195:24195:0115/173332.006377:WARNING:gpu_video_decode_accelerator_factory.cc(246)]
   : Initializing VAAPI VDA.
   -
   [24195:24195:0115/173332.061903:WARNING:gpu_video_decode_accelerator.cc(397)]
   : Created VDA
   - [24195:24195:0115/173332.091980:ERROR:vaapi_wrapper.cc(1467)] :
vaCreateConfig
   failed VA error: the requested VAProfile is not supported
   - [24195:24195:0115/173332.092033:ERROR:vaapi_wrapper.cc(848)] : Failed
   to create VaapiWrapper for va_profile: -1



Am Di., 15. Jan. 2019 um 17:31 Uhr schrieb David Schuster <
spearhead2k3@gmail.com>:

**Comment by saiarcot895** [comment]
@omen23 What video are you using to test GPU video decoding? Additionally, can you try using the [h264ify](https://chrome.google.com/webstore/detail/h264ify/aleakchihdccplidncghkekgioiakgal) extension, which tells the browser that only h264 videos are supported? This should be supported for hardware video decoding by most GPUs, and so should work on Nvidia.

**Comment by biswasab** [comment]
@omen23 If you're talking about my patch, It will work but it is disabled by default. You have to enable chrome://flags/#ignore-gpu-blacklist to get it working. Normally it will require a patched libva-vdpau-driver so I have disabled it.

**Comment by omen23** [comment]
i got all this enabled and VA acceleration is broken in the new version

Am Di., 15. Jan. 2019 um 18:30 Uhr schrieb Akarshan Biswas <
notifications@github.com>:

**Comment by biswasab** [comment]
@omen23 Are you using my patch?

**Comment by omen23** [comment]
just normal vaapi GpuVideoDecoder this is the error

Log Messages

   -
   [2964:2964:0116/134243.695751:WARNING:gpu_video_decode_accelerator.cc(383)]
   : Created the VDA factory
   -
   [2964:2964:0116/134243.695790:WARNING:gpu_video_decode_accelerator_factory.cc(246)]
   : Initializing VAAPI VDA.
   -
   [2964:2964:0116/134243.727730:WARNING:gpu_video_decode_accelerator.cc(397)]
   : Created VDA
   - [2964:2964:0116/134243.729766:ERROR:vaapi_wrapper.cc(1467)] :
vaCreateConfig
   failed VA error: the requested VAProfile is not supported
   - [2964:2964:0116/134243.729820:ERROR:vaapi_wrapper.cc(848)] : Failed to
   create VaapiWrapper for va_profile: -1


Am Mi., 16. Jan. 2019 um 13:40 Uhr schrieb Akarshan Biswas <
notifications@github.com>:

**Comment by omen23** [comment]
and yes im using h264ify i even wrote a tutorial for people on how to use
GPU acceleration and mentioned your patched chromium build
 https://github.com/omen23/ffmpeg-ffnvcodec-explanation

Hope you can fix the error .. somewhere in vaapi_wrapper.cc something
returns the wrong value and yes this is my command line
/usr/lib/chromium-browser/chromium-browser
--ppapi-flash-path=/usr/lib/pepperflashplugin-nonfree/libpepflashplayer.so
--ppapi-flash-version=32.0.0.101 --enable-pinch --flag-switches-begin
--enable-accelerated-video --enable-gpu-rasterization --enable-zero-copy
--ignore-gpu-blacklist
--enable-features=BloatedRendererDetection,VizDisplayCompositor
--flag-switches-end
--file-url-path-alias=/gen=/usr/lib/chromium-browser/gen

Am Mi., 16. Jan. 2019 um 13:46 Uhr schrieb David Schuster <
spearhead2k3@gmail.com>:

**Comment by omen23** [comment]
maybe we wont need the patched builds
<https://github.com/chromium/chromium/commit/31225b9c5f3f685d65f742dc129241c30c32157c>
anymore in the near future

Am Mi., 16. Jan. 2019 um 13:51 Uhr schrieb David Schuster <
spearhead2k3@gmail.com>:

**Comment by omen23** [comment]
like my icon John D. Carmack said .. nothing better than good old printf
shotgun debugging ...
and we got the line numbers :
Log Messages

   -
   [2964:2964:0116/134243.695751:WARNING:gpu_video_decode_accelerator.cc(383)]
   : Created the VDA factory
   -
   [2964:2964:0116/134243.695790:WARNING:gpu_video_decode_accelerator_factory.cc(246)]
   : Initializing VAAPI VDA.
   -
   [2964:2964:0116/134243.727730:WARNING:gpu_video_decode_accelerator.cc(397)]
   : Created VDA
   - in the version im running now log stops here
73.0.3642.0-0ubuntu1~ppa4~18.10.1
   - in the two updated builds these two lines are also displayed and
   GpuVideoDecoder fails
   - [2964:2964:0116/134243.729766:ERROR:vaapi_wrapper.cc(1467)] :
vaCreateConfig
   failed VA error: the requested VAProfile is not supported
   - [2964:2964:0116/134243.729820:ERROR:vaapi_wrapper.cc(848)] : Failed to
   create VaapiWrapper for va_profile: -1

yes the error happens here
849 VAProfile va_profile = ProfileToVAProfile(profile, mode); Kind Regards

Am Mi., 16. Jan. 2019 um 16:55 Uhr schrieb Saikrishna Arcot <
notifications@github.com>:

**Comment by omen23** [comment]
You only have to figure out which value corresponds to none
"maybe" VAProfileNone, if not which function returns -1 as value for
va_profile - because the va_profiles are either function pointers to VA
entrypoints but i believe just integers to the mode. like
vainfo
libva info: VA-API version 1.2.0
libva info: va_getDriverName() returns 0
libva info: Trying to open
/usr/lib/x86_64-linux-gnu/dri/nvidia_drv_video.so
libva info: Found init function __vaDriverInit_1_2
libva info: va_openDriver() returns 0
vainfo: VA-API version: 1.2 (libva 2.2.0)
vainfo: Driver version: Splitted-Desktop Systems VDPAU backend for VA-API -
0.7.4
vainfo: Supported profile and entrypoints
     VAProfileMPEG2Simple            : VAEntrypointVLD
     VAProfileMPEG2Main              : VAEntrypointVLD
     VAProfileMPEG4Simple            : VAEntrypointVLD
     VAProfileMPEG4AdvancedSimple    : VAEntrypointVLD
     <unknown profile>               : VAEntrypointVLD
     VAProfileH264Main               : VAEntrypointVLD
     VAProfileH264High               : VAEntrypointVLD
     VAProfileVC1Simple              : VAEntrypointVLD
     VAProfileVC1Main                : VAEntrypointVLD
     VAProfileVC1Advanced            : VAEntrypointVLD

VAProfileH264Main is probably 5 or something like that.

Kind Regards

Am Mi., 16. Jan. 2019 um 20:17 Uhr schrieb David Schuster <
spearhead2k3@gmail.com>:

**Comment by omen23** [comment]
Sorry .. yeah you were right ... i didnt read your first sentence ...
maybe the "<unknown profile>" which is the DivX entrypoint is the problem

Kind Regards

Am Mi., 16. Jan. 2019 um 20:21 Uhr schrieb David Schuster <
spearhead2k3@gmail.com>:

**Comment by omen23** [comment]
he said it .. the problem with "high level C or C++ programming is that
most people dont fully understand the states their software might be in" ..
and thats sooo valid ...
why is va_profile not getting the correct value ... its cool that you add
debugging for the gpu process still id log it all and check line for line
where its not able to get the right value assigned to va_profile.

id even be willing to run the build on my machine if it helps.

kind regards

Am Mi., 16. Jan. 2019 um 20:25 Uhr schrieb David Schuster <
spearhead2k3@gmail.com>:

**Comment by omen23** [comment]
i hope you didnt get me wrong ... it is thanks to your work that GPU
acceleration works but "how tf are they able to break it all the time"?

Kind Regards


Am Mi., 16. Jan. 2019 um 20:32 Uhr schrieb David Schuster <
spearhead2k3@gmail.com>:

**Comment by omen23** [comment]
well do you want to enable libva via MojoVideoDecoder then or still use
GpuVideoDecoder? tbh as long as the GPU acceleration works im happy ... im
still using 73.0.3642.0-0ubuntu1~ppa4~18.10.1

Kind Regards

Am Do., 17. Jan. 2019 um 17:25 Uhr schrieb Saikrishna Arcot <
notifications@github.com>:

**Comment by saiarcot895** [comment]
@Queuecumber Yes, it should be working again.

@omen23, did you test with a Youtube video as well (not just a DivX video)?

**Comment by omen23** [comment]
with h264ify
it still fails with Log Messages

   -
   [12340:12340:0123/113240.121285:WARNING:gpu_video_decode_accelerator.cc(383)]
   : Created the VDA factory
   -
   [12340:12340:0123/113240.121318:WARNING:gpu_video_decode_accelerator_factory.cc(246)]
   : Initializing VAAPI VDA.
   -
   [12340:12340:0123/113240.158929:WARNING:gpu_video_decode_accelerator.cc(397)]
   : Created VDA
   -
   [12340:12340:0123/113240.161296:WARNING:gpu_video_decode_accelerator.cc(383)]
   : Created the VDA factory
   -
   [12340:12340:0123/113240.161331:WARNING:gpu_video_decode_accelerator_factory.cc(246)]
   : Initializing VAAPI VDA.
   -
   [12340:12340:0123/113240.161474:WARNING:gpu_video_decode_accelerator.cc(397)]
   : Created VDA
   - [12340:12340:0123/113240.189316:ERROR:vaapi_wrapper.cc(1467)] :
vaCreateConfig
   failed VA error: the requested VAProfile is not supported
   - [12340:12340:0123/113240.189375:ERROR:vaapi_wrapper.cc(848)] : Failed
   to create VaapiWrapper for va_profile: -1
   - [12340:12340:0123/113240.247568:ERROR:vaapi_wrapper.cc(1467)] :
vaCreateConfig
   failed VA error: the requested VAProfile is not supported
   - [12340:12340:0123/113240.247611:ERROR:vaapi_wrapper.cc(848)] : Failed
   to create VaapiWrapper for va_profile: -1



I am sticking to 73.0.3642.0-0ubuntu1~ppa4~18.10.1.
Hope the error gets fixed soon.

Kind Regards


Am Mi., 23. Jan. 2019 um 11:28 Uhr schrieb David Schuster <
spearhead2k3@gmail.com>:

**Comment by omen23** [comment]
Will you ever fix the VAAPI problem on nvidia cards?

Kind Regards

Am Mi., 23. Jan. 2019 um 11:34 Uhr schrieb David Schuster <
spearhead2k3@gmail.com>:

**Comment by kingio** [comment]
@omen23 he is doing this for free, you could donate him so he keeps motivated.

**Comment by omen23** [comment]
i can help for free (=

Am Mo., 28. Jan. 2019 um 10:03 Uhr schrieb Carlos Domínguez <
notifications@github.com>:

**Comment by Queuecumber** [comment]
Doesn't Nvidia have vdpau or something different?

**Comment by omen23** [comment]
theres vdpau-va-driver

Am Mo., 28. Jan. 2019 um 13:32 Uhr schrieb Max Ehrlich <
notifications@github.com>:

**Comment by biswasab** [comment]
@omen23 It will not work especially on newer GPUs. I have tested it. It's Nvidia's fault. Please ask them to add support for vaapi or you can buy an intel/amd card for yourself.

**Comment by saiarcot895** [comment]
@omen23 I'm waiting on Chromium for a source tarball that's actually buildable. The last two dev releases have had at least one missing file in the tarball.

@biswasab Is this the case even with a patched vdpau-va-driver?

**Comment by omen23** [comment]
@biswasab https://github.com/chromium/chromium/commit/31225b9c5f3f685d65f742dc129241c30c32157c
maybe this will fix it soon in the official repos? official ubuntu repos still stuck at 71.x ... like I said i dont got a problem with waiting ...
@saiarcot895 using a 73.x version with working GPU acceleration `73.0.3642.0-0ubuntu1~ppa4~18.10.1`
wish you the best and dont let people spread hate - the one guy just posted to make us all feel uncomfortable

**Comment by biswasab** [comment]
@saiarcot895 Yes. I used the vdpau-va-driver  with VaQuerySurfaceAttributes patch.

@omen23  I have tried that patch as well. It's the error of the vaapi-->vdpau bridge(vdpau-va-driver)(I think).

**Comment by omen23** [comment]
i still wonder it works in 73.0.3642.0-0ubuntu1~ppa4~18.10.1 but not
73.0.3664.3-0ubuntu1~ppa2~18.10.1...
what has changed in the vaapi_wrapper

Am Di., 29. Jan. 2019 um 07:35 Uhr schrieb Akarshan Biswas <
notifications@github.com>:

**Comment by elmeyer** [comment]
For those of us who accidentally upgraded without being aware of this issue, does anyone have the 73.0.3642.0-0ubuntu1~ppa4 debs cached and could provide them here for rolling back? e.g. @omen23
TIA!

**Comment by saiarcot895** [comment]
@elmeyer Are you on Nvidia or AMD?

**Comment by elmeyer** [comment]
@saiarcot895 I‘m on an Intel GMA X4500HD (G45 chipset) using self-compiled VAAPI libs from here: https://bitbucket.org/alium/g45-h264/downloads/

VAAPI on Chromium did work before upgrading, and it continues to work using mpv. Kubuntu 18.10.1 amd64.

**Comment by omen23** [comment]
Hey guys,
I host them here:
https://openload.co/f/yHhF3W0OmPY/chromium_73.0.3642.0-0ubuntu1-ppa4.rar

Thanks tho no worries, David

Am Mo., 11. Feb. 2019 um 13:40 Uhr schrieb Lars Meyer <
notifications@github.com>:

**Comment by omen23** [comment]
> For those of us who accidentally upgraded without being aware of this issue (and then cleared their apt cache out of a bad habit...), does anyone have the 73.0.3642.0-0ubuntu1~ppa4 debs cached and could provide them here for rolling back? e.g. @omen23
> TIA!

Hey guys,
I host them here: 
https://openload.co/f/yHhF3W0OmPY/chromium_73.0.3642.0-0ubuntu1-ppa4.rar
(contains
chromium-browser_73.0.3642.0-0ubuntu1~ppa4~18.10.1_amd64.deb
chromium-browser-l10n_73.0.3642.0-0ubuntu1~ppa4~18.10.1_all.deb
chromium-codecs-ffmpeg-extra_73.0.3642.0-0ubuntu1_ppa4_18.10.1_amd64.deb
)

Thanks tho no worries, David

**Comment by saiarcot895** [comment]
@omen23 The PPA has been updated with a build with more debugging symbols, and this version should be able to print an error message if it wasn't able to map a `VAProfile`.

**Comment by omen23** [comment]
I am upgrading rn ..
I will launch from terminal to tell you exactly what is on about:gpu and
the terminal..
But before I guess I'll grab some food.

Greetings

Am Do., 14. Feb. 2019 um 02:27 Uhr schrieb Saikrishna Arcot <
notifications@github.com>:

**Comment by omen23** [comment]
Hey,
so about:gpu gives me this 
```
Log Messages
[14446:14446:0214/122531.456218:WARNING:gpu_video_decode_accelerator_factory.cc(246)] : Initializing VAAPI VDA.
[14446:14446:0214/122531.501579:ERROR:vaapi_wrapper.cc(1472)] : vaCreateConfig failed VA error: the requested VAProfile is not supported
[14446:14446:0214/122531.501617:ERROR:vaapi_wrapper.cc(853)] : Failed to create VaapiWrapper for va_profile: -1
```
and in Terminal I get this:
```
Failed to create dedicated X11 display! # I guess this isn't really the bug this message just pops up randomly
[14446:14446:0214/122531.501579:ERROR:vaapi_wrapper.cc(1472)] vaCreateConfig failed VA error: the requested VAProfile is not supported
[14446:14446:0214/122531.501617:ERROR:vaapi_wrapper.cc(853)] Failed to create VaapiWrapper for va_profile: -1
[1:15:0214/122531.501772:ERROR:render_media_log.cc(30)] MediaEvent: MEDIA_ERROR_LOG_ENTRY {"error":"VDA Error 4"}
```
I will roll back now ... cause I need this as production system atm ...

**Comment by Queuecumber** [comment]
I'm on `73.0.3683.10` as well intel hardware. Computer sounded loud while playing a video the other day and when I checked media-internals it shows MojoVideoDecoder, I know thats the new thing, is that hardware accelerated or not? 

about:gpu showed

```
[3988:3988:0217/155639.023700:ERROR:sandbox_linux.cc(364)] : InitializeSandbox() called with multiple threads in process gpu-process.
[3988:3988:0217/155648.973418:WARNING:gpu_video_decode_accelerator_factory.cc(246)] : Initializing VAAPI VDA.
```

e.g. nothing suspicious

**Comment by saiarcot895** [comment]
It is (very likely) using GPU. The second line (initializing VAAPI VDA) shows that the GPU decoder got initialized, and `intel_gpu_top` on my system shows that it is using the GPU.

**Comment by Queuecumber** [comment]
How are you interpreting `intel_gpu_top` to mean its using the video decoder? I remember this being debated at some point in the past but I'm not sure what the conclusion was. Also is there a way to disable Mojo and use Gpu so that I can compare?

**Comment by saiarcot895** [comment]
@Queuecumber You can run `chromium-browser --disable-features=MojoVideoDecoder` from the command line; that'll disable the use of Mojo.

As for `intel_gpu_top`, I tested by pausing and resuming a Youtube video while looking at `intel_gpu_top`, and determined that the GPU usage appeared to be when the video was playing. I wasn't moving my mouse or doing any other activity when measuring the GPU usage.

**Comment by Queuecumber** [comment]
yeah but it takes GPU time to render the frame too

Anyway I did a very unscientific test to try to figure this out by using the flag to disable all hardware encoding, then disabling only mojo, then using mojo and looking at intel GPU top for GPU activity and powertop for power consumption. I couldnt tell the difference between all three in either program. But it's pretty hard to control for all the variables.

**Comment by omen23** [comment]
`74.0.3710.0-0ubuntu1~ppa6~18.10.1`
```
 chromium-browser 
[19726:19751:0301/185539.242280:ERROR:simple_index_file_posix.cc(61)] Could not get file info for /home/$USER/.config/chromium/Default/Service Worker/ScriptCache/0a53320a35220130_1
[19726:19751:0301/185539.242330:ERROR:simple_index_file_posix.cc(61)] Could not get file info for /home/$USER/.config/chromium/Default/Service Worker/ScriptCache/88982711b7d82f98_0
Failed to create dedicated X11 display!
[19767:19767:0301/185551.209746:ERROR:vaapi_wrapper.cc(1458)] vaCreateConfig failed VA error: the requested VAProfile is not supported
[19767:19767:0301/185551.209784:ERROR:vaapi_wrapper.cc(851)] Failed to create VaapiWrapper for va_profile: -1
[1:28:0301/185551.209910:ERROR:render_media_log.cc(27)] MediaEvent: MEDIA_ERROR_LOG_ENTRY {"error":"VDA Error 4"}
```

**Comment by JurgenWolf** [comment]
I have the same issue with latest `74.0.3710.0-0ubuntu1~ppa6~19.04.1 amd64` running on Debian unstable with GeForce GT 610 @ NVIDIA 390.116 (recent upgrade from 390.87 driver and Chromium 72 @saiarcot895 Beta PPA).
`vdpau-va-driver 0.7.4-7ubuntu1~ppa1~18.10.1` is installed as well.
Before this upgrade were no any problems with video decode.

> [3044:3044:0304/235840.608853:WARNING:gpu_video_decode_accelerator_factory.cc(246)] : Initializing VAAPI VDA.
[3044:3044:0304/235840.671851:ERROR:vaapi_wrapper.cc(1458)] : vaCreateConfig failed VA error: the requested VAProfile is not supported
[3044:3044:0304/235840.672187:ERROR:vaapi_wrapper.cc(851)] : Failed to create VaapiWrapper for va_profile: -1

> $ vainfo
libva info: VA-API version 1.4.0
libva info: va_getDriverName() returns 0
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/nvidia_drv_video.so
libva info: Found init function __vaDriverInit_1_2
libva info: va_openDriver() returns 0
vainfo: VA-API version: 1.4 (libva 2.4.0)
vainfo: Driver version: Splitted-Desktop Systems VDPAU backend for VA-API - 0.7.4
vainfo: Supported profile and entrypoints
      VAProfileMPEG2Simple            :	VAEntrypointVLD
      VAProfileMPEG2Main              :	VAEntrypointVLD
      VAProfileMPEG4Simple            :	VAEntrypointVLD
      VAProfileMPEG4AdvancedSimple    :	VAEntrypointVLD
      unknown profile               :	VAEntrypointVLD
      VAProfileH264Main               :	VAEntrypointVLD
      VAProfileH264High               :	VAEntrypointVLD
      VAProfileVC1Simple              :	VAEntrypointVLD
      VAProfileVC1Main                :	VAEntrypointVLD
      VAProfileVC1Advanced            :	VAEntrypointVLD

**Comment by JurgenWolf** [comment]
I also tested it using `--disable-features=MojoVideoDecoder`
No any effect.

**Comment by JurgenWolf** [comment]
And in terminal I get one more error:

> [4883:4893:0305/014120.455889:ERROR:render_media_log.cc(27)] MediaEvent: MEDIA_ERROR_LOG_ENTRY {"error":"VDA Error 4"}

**Comment by saiarcot895** [comment]
I don't have a fix for this yet. What I can tell you, however, is that Chromium is doing the same thing on both Intel and Nvidia GPUs. Specifically, Chromium calls `vaCreateConfig` with a profile of -1 (None) on both GPUs. The Intel libva implementation supports it, while the Nvidia libva implementation does not.

**Comment by omen23** [comment]
Hey,
how did you get libva 2.4.0 on 18.10 ?
It is not yet supported and upstream is experimental.

Kind Regards

Am Di., 5. März 2019 um 04:43 Uhr schrieb Saikrishna Arcot <
notifications@github.com>:

**Comment by JurgenWolf** [comment]
As already said above, I'm on Debian sid/unstable.
So libva is from Debian repository.
And it worked prior this recent upgrade of Chromium.

**Comment by saiarcot895** [comment]
@omen23 (and anyone else hitting this issue on Nvidia), can you install and test the (relevant) packages?

http://vps.saiarcot895.cloudns.pro/chromium-browser_74.0.3710.0-0ubuntu1~ppa7~18.10.1_amd64.deb
http://vps.saiarcot895.cloudns.pro/chromium-codecs-ffmpeg_74.0.3710.0-0ubuntu1~ppa7~18.10.1_amd64.deb
http://vps.saiarcot895.cloudns.pro/chromium-codecs-ffmpeg-extra_74.0.3710.0-0ubuntu1~ppa7~18.10.1_amd64.deb

**Comment by dagraver** [comment]
It works for me on a GTX-770 with some work:

1) It needs the vdpau-va-driver with implement-vaquerysurfaceattributes.patch (obviously)
2) I have to run chromium with --disable-gpu-sandbox otherwise this happens and acceleration fails:

libva error: /usr/lib/x86_64-linux-gnu/dri/nvidia_drv_video.so init failed
[19632:19632:0305/171902.782877:ERROR:vaapi_wrapper.cc(325)] vaInitialize failed (ignore if using Wayland desktop environment): unknown libva error
libva error: va_getDriverName() failed with unknown libva error,driver_name=(null)
[19632:19632:0305/171902.782962:ERROR:vaapi_wrapper.cc(333)] vaInitialize failed using DRM: unknown libva error
[19632:19632:0305/171902.782980:ERROR:vaapi_wrapper.cc(851)] Failed to create VaapiWrapper for va_profile: 7
[19632:19632:0305/171902.782998:ERROR:gpu_video_decode_accelerator_factory.cc(187)] Initialization of one or more VDAs failed.
[19601:19601:0305/171908.082353:ERROR:http_bridge.cc(127)] Not implemented reached in virtual void syncer::HttpBridgeFactory::OnSignalReceived()

Thanks man, for spending so much time on this!

**Comment by JurgenWolf** [comment]
Sandbox was always completely disabled on my machine via `--no-sandbox`
But this doesn't help with Chromium 73/74.

**Comment by pgiri** [comment]
> @omen23 (and anyone else hitting this issue on Nvidia), can you install and test the (relevant) packages?
> 

@saiarcot895 I have nvidia and never managed to get GPU acceleration working. With above packages and using ``--disable-features=MojoVideoDecoder``, I have it working: In 'media-internals', I get "video_decoder | GpuVideoDecode" and playing a youtube video shows CPU usage is about 15% (which used to be about 50% before). Thank you very much!

**Comment by JurgenWolf** [comment]
I can confirm.
With ppa7 build it works for me as expected! Even without `--disable-features=MojoVideoDecoder`
Tested at 1080p.
Thank you!!!

**Comment by saiarcot895** [comment]
Thanks for the reports! I'm not going to push this build to the PPA, but when this week's release for the dev and beta channels happen from Chromium, I'll include this fix on both (since at this point, beta is broken on Nvidia GPUs as well).

@pgiri `MojoVideoDecoder` might use the GPU behind the scenes. `MojoVideoDecoder` is kind of a wrapper around some of the current video decoders in Chromium to reduce the IPC overhead, so when you see `MojoVideoDecoder`, it may or may not be using the GPU (you'd have to check with the CPU usage and `intel_gpu_top` if you're using Intel GPUs).

**Comment by dagraver** [comment]
I can confirm the Mojo decoder is indeed utilising the Nvidia GPU, so probably no need to disable that. Nvidia users can check that by observing Video Engine Utilisation in theNvidia X Server Settings program.

**Comment by saiarcot895** [comment]
@dagraver With Chromium open and GPU video decoding working, can you give me the output of `lsof /dev/nvidia*`?

**Comment by dagraver** [comment]
@saiarcot895 Sure, here it is:

```COMMAND          PID   USER   FD   TYPE  DEVICE SIZE/OFF NODE NAME
chromium-browse 7558 graver  mem    CHR 195,255           631 /dev/nvidiactl
chromium-browse 7558 graver  mem    CHR   195,0           632 /dev/nvidia0
chromium-browse 7558 graver  122u   CHR 195,255      0t0  631 /dev/nvidiactl
chromium-browse 7558 graver  124u   CHR 195,254      0t0  783 /dev/nvidia-modeset
chromium-browse 7558 graver  125u   CHR   195,0      0t0  632 /dev/nvidia0
chromium-browse 7558 graver  126u   CHR   195,0      0t0  632 /dev/nvidia0
chromium-browse 7558 graver  129u   CHR 195,254      0t0  783 /dev/nvidia-modeset
chromium-browse 7558 graver  130u   CHR   195,0      0t0  632 /dev/nvidia0
chromium-browse 7558 graver  131u   CHR   195,0      0t0  632 /dev/nvidia0
chromium-browse 7558 graver  132u   CHR   195,0      0t0  632 /dev/nvidia0
chromium-browse 7558 graver  133u   CHR   195,0      0t0  632 /dev/nvidia0
chromium-browse 7558 graver  134u   CHR   195,0      0t0  632 /dev/nvidia0
chromium-browse 7558 graver  164u   CHR 195,254      0t0  783 /dev/nvidia-modeset
chromium-browse 7558 graver  167u   CHR   195,0      0t0  632 /dev/nvidia0
chromium-browse 7558 graver  168u   CHR   195,0      0t0  632 /dev/nvidia0
chromium-browse 7558 graver  169u   CHR   195,0      0t0  632 /dev/nvidia0
chromium-browse 7558 graver  170u   CHR   195,0      0t0  632 /dev/nvidia0
chromium-browse 7558 graver  171u   CHR   195,0      0t0  632 /dev/nvidia0
chromium-browse 7558 graver  188u   CHR 195,255      0t0  631 /dev/nvidiactl
chromium-browse 7558 graver  189u   CHR   195,0      0t0  632 /dev/nvidia0
chromium-browse 7558 graver  190u   CHR   195,0      0t0  632 /dev/nvidia0
chromium-browse 7558 graver  191u   CHR   195,0      0t0  632 /dev/nvidia0
chromium-browse 7558 graver  192u   CHR 195,254      0t0  783 /dev/nvidia-modeset
chromium-browse 7558 graver  193u   CHR   195,0      0t0  632 /dev/nvidia0
chromium-browse 7558 graver  194u   CHR   195,0      0t0  632 /dev/nvidia0
chromium-browse 7558 graver  195u   CHR   195,0      0t0  632 /dev/nvidia0
chromium-browse 7558 graver  196u   CHR   195,0      0t0  632 /dev/nvidia0
chromium-browse 7558 graver  197u   CHR   195,0      0t0  632 /dev/nvidia0
chromium-browse 7558 graver  198u   CHR   195,0      0t0  632 /dev/nvidia0
chromium-browse 7558 graver  199u   CHR   195,0      0t0  632 /dev/nvidia0
```

**Comment by omen23** [comment]
fixes it Clap
Saikrishna Arcotyou are the man - please mail me the lines you had to
change im interested
thanks man

Am Mi., 6. März 2019 um 08:00 Uhr schrieb dagraver <notifications@github.com

**Comment by omen23** [comment]
I did not disable MojoVideoDecoder as it uses the GPU ...

Am Mi., 6. März 2019 um 10:26 Uhr schrieb David Schuster <
spearhead2k3@gmail.com>:

**Comment by omen23** [comment]
about:gpu log
```
Log Messages
[27544:27544:0306/102450.444008:WARNING:x11_util.cc(1423)] : X error received: serial 367, error_code 3 (BadWindow), request_code 4, minor_code 0 (Unknown)
[27544:27694:0306/102450.525752:WARNING:x11_util.cc(1423)] : X error received: serial 511, error_code 3 (BadWindow), request_code 4, minor_code 0 (Unknown)
[27544:27544:0306/102459.203056:WARNING:gpu_video_decode_accelerator_factory.cc(246)] : Initializing VAAPI VDA.
[27544:27544:0306/102531.744350:WARNING:x11_util.cc(1423)] : X error received: serial 1274, error_code 3 (BadWindow), request_code 4, minor_code 0 (Unknown)
[27544:27694:0306/102532.009380:WARNING:x11_util.cc(1423)] : X error received: serial 3589, error_code 3 (BadWindow), request_code 4, minor_code 0 (Unknown)
[27544:27544:0306/102738.399344:WARNING:gpu_video_decode_accelerator_factory.cc(246)] : Initializing VAAPI VDA.
[27544:27544:0306/102744.706969:WARNING:ipc_message_attachment_set.cc(49)] : MessageAttachmentSet destroyed with unconsumed attachments: 0/1
[27544:27544:0306/102809.502850:WARNING:x11_util.cc(1423)] : X error received: serial 1734, error_code 3 (BadWindow), request_code 4, minor_code 0 (Unknown)
[27544:27694:0306/102811.193662:WARNING:x11_util.cc(1423)] : X error received: serial 7534, error_code 3 (BadWindow), request_code 4, minor_code 0 (Unknown)
[27544:27544:0306/102911.128075:WARNING:gpu_video_decode_accelerator_factory.cc(246)] : Initializing VAAPI VDA.
```
terminal output:
```
$ chromium-browser 
Failed to create dedicated X11 display!
```
my command-line: `/usr/lib/chromium-browser/chromium-browser --ppapi-flash-path=/usr/lib/pepperflashplugin-nonfree/libpepflashplayer.so --ppapi-flash-version=32.0.0.142 --enable-pinch --flag-switches-begin --enable-gpu-rasterization --enable-zero-copy --ignore-gpu-blacklist --enable-features=BloatedRendererDetection --flag-switches-end --file-url-path-alias=/gen=/usr/lib/chromium-browser/gen`

Thanks again man :+1:

**Comment by seppokoo** [comment]
On 18.04.1 this update broke all GPU accelerations: 
ERROR:gl_surface_glx.cc(77)] XGetWindowAttributes failed for window 12582916.
ERROR:gl_surface_glx.cc(809)] Failed to get GLXConfig
ERROR:gpu_info_collector.cc(51)] gl::GLContext::CreateOffscreenGLSurface failed
ERROR:gpu_info_collector.cc(181)] Could not create surface for info collection.
ERROR:gpu_init.cc(62)] gpu::CollectGraphicsInfo failed.
ERROR:viz_main_impl.cc(170)] Exiting GPU process due to errors during initialization

Nvidia Quadro P400 driver 390.77

**Comment by JurgenWolf** [comment]
What update exactly?

For me this ppa7 build (see download links above) on Debian sid/unstable with 390.116 driver - works smoothly even at 60FPS.

**Comment by JurgenWolf** [comment]
My command line:
`--no-sandbox --disable-infobars --disable-logging --disable-renderer-accessibility --process-per-site --flag-switches-begin --disable-threaded-scrolling --enable-gpu-rasterization --enable-quic --enable-zero-copy --ignore-gpu-blacklist --disable-site-isolation-trials --disable-smooth-scrolling --disable-features=NativeNotifications --flag-switches-end`

**Comment by JurgenWolf** [comment]
I just tested this new upload to Dev PPA `74.0.3724.8-0ubuntu1~ppa1~18.10.1`
Degraded performance and various distortions like horizontal lines during video playback etc.

**Comment by JurgenWolf** [comment]
From my experience, Dev branch is always with "surprises"...
Waiting for Beta + Fix...
Thank you.

**Comment by omen23** [comment]
for me `74.0.3724.8-0ubuntu1~ppa1~18.10.1` works great!

**Comment by JurgenWolf** [comment]
Probably, new upstream release doesn't affect all GPUs... but with GeForce GT 610 and 390.116 it works very bad.

**Comment by omen23** [comment]
drivers too old - sucks for legacy products

**Comment by JurgenWolf** [comment]
390.116 is not old... it was released by Nvidia upstream ~1 week ago...
And 390 serie is the last serie which supports Fermi GPUs like GF GT 610.

**Comment by JurgenWolf** [comment]
74.0.3710.0 (with fix) - works good at 390.116
74.0.**3724.8** (with fix) - absolutely another impression... slowness and distortions...

So issues are related to Chromium74 3724 branch definitely... something important was added/removed/recoded there...

**Comment by omen23** [comment]
@JurgenWolf where did you get the l10n package for the 3710 vesion?

**Comment by JurgenWolf** [comment]
I didn't get it.
l10n package is optional/recommended. I use Chromium without this package.

**Comment by saiarcot895** [comment]
I'm closing this bug, as the thread history has gotten long enough, and the original issue was fixed. @JurgenWolf please open a new issue.

**Comment by omen23** [comment]
Hello,
thanks again for putting in that much effort for people with Intel and
Nvidia GPUs :+1:
Regards David

Am Fr., 8. März 2019 um 16:57 Uhr schrieb Saikrishna Arcot <
notifications@github.com>:
